### ISSUE TITLE
Middleware appears to be returning odd result

### ISSUE THREAD

**Issue by speterson-zoll** [khellang/Middleware#123]
@khellang:

Hello again.  I have an API that I configured with your ProblemDetails middleware, but I'm getting curious results when I have a validation error.  I have a controller method that has two required parameters, both of which have the [Required] attribute tags:

```
[HttpGet("{wearId}/alerts/{alertDefinitionType}/settings")]
[ProducesResponseType(typeof(AlertSettingsDto), 200)]
public async Task<IActionResult> GetAlertSettings([FromRoute] [Required] string itemId,
  [FromRoute] [Required] AlertType alertType)
{
    Logger.LogDebug("wearId = {ItemId}", itemId);
    Logger.LogDebug("alertType = {AlertType}", alertType);

    var settingsData =
      await _alertSettingsService.GetAlertSettings(itemId, alertType);

    if (settingsData == null)
    {
      return (IActionResult)Ok(settingsData);
    }

    if (settingsData.AlertStatusCode != AlertStatusCode.Success)
    {
      return BadRequest(EnumHelper.GetEnumDescription(settingsData.AlertStatusCode));
    }

    return Ok(settingsData);
```

Running this code via Postman (with an empty itemId) WITHOUT your middleware returns a 400 Bad Request with the following Body:

```
{
    "type": "https://tools.ietf.org/html/rfc7231#section-6.5.1",
    "title": "One or more validation errors occurred.",
    "status": 400,
    "traceId": "|78909fb6-4db110476feba5a1.",
    "errors": {
        "itemId": [
            "The itemId field is required."
        ]
    }
}
```

I enabled your Middleware (5.2.3) in my API like so:

```
services.AddControllers().AddProblemDetailsConventions();
services.AddProblemDetails();

app.UseProblemDetails();

```

Running via Postman returns a 200 OK response with this body:

```
{
    "type": "https://httpstatuses.com/422",
    "title": "One or more validation errors occurred.",
    "status": 422,
    "traceId": "|1c3e47fe-4c22ca390e325aa6.",
    "correlationId": "00000000-0000-0000-0000-000000000000",
    "errors": {
        "itemId": [
            "The itemId field is required."
        ]
    }
}
```

I like that the second example has 422 as a status code, as it's more accurate.  However, the 200 OK response is confusing my rest client, because it thinks things are OK.  Is this expected behavior?  I thought I was going to get a 422, not a 200.

**Comment by khellang** [comment]
Hmm, yeah, that's weird. It's definitely supposed to be 422. I'll look into it 😅

**Comment by khellang** [comment]
I was able to reproduce it pretty quicky with some unit tests. Turns out it's an ASP.NET Core 3.1-specific issue that's been addressed in .NET 5. I've pushed a new version, v5.2.4, to NuGet. Hopefully that should be the last snag - at least all the new tests are green 😅

**Comment by khellang** [comment]
Looks like it was actually considered backporting the fix to 3.1, but was decided against... https://github.com/dotnet/aspnetcore/issues/17568 😞

**Comment by speterson-zoll** [comment]
Very nice, thanks again for your help.  I'll be putting this version through its paces today.
